taichi/rhi/device.h

```diff
@@ -597,14 +556,30 @@ struct SurfaceConfig {
   uint32_t height{1};
 };
 
-struct TI_DLL_EXPORT ImageParams {
+enum class ImageAllocUsage : int {
```

Comment:
Maybe we should build a `enum flags` class template?

Reply:
Sounds good, will get it refactorized later.